tests/integration_tests.rs

```diff
@@ -0,0 +1,120 @@
+#[cfg(test)]
```

Comment:
I had to move the tests here because I was hitting some build issues with the Rust action.

This ensures that the `function-runner` binary is compiled for the integration tests.

The action runs the command `cargo test --bin function_runner`. When the command is run `cargo` isn't  able to determine the dependency on the `function-runner` binary and panics. This can be simulated with the previous version of the code and running `cargo clean && cargo test --bin function_runner`